### PR TITLE
feat: Highlighting matching chars

### PR DESCRIPTION
**Pull request by dhananjay0601**
# Description

Highlighting matching characters based on searched keyword 

Fixes https://github.com/devtron-labs/dashboard/issues/786

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Tested by typing keywords in the search input box.

# Checklist:

* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [X] I have performed a self-review of my own code
* [X] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/ResourceBrowser/ResourceList/K8SResourceList.tsx [hunk 1]
```diff
@@ -17,6 +17,7 @@ import { toast } from 'react-toastify'
 import { EventList } from './EventList'
 import Tippy from '@tippyjs/react'
 import ResourceFilterOptions from './ResourceFilterOptions'
+import { log } from 'console'
```
Comment (shivani170) [inline src/components/ResourceBrowser/ResourceList/K8SResourceList.tsx:20]: remove this

File: src/components/common/helpers/Helpers.tsx [hunk 2]
```diff
@@ -1207,3 +1207,14 @@ export const k8sStyledAgeToSeconds = (duration: string): number => {
 export const eventAgeComparator = <T,>(key: string): any => {
     return (a: T, b: T) => k8sStyledAgeToSeconds(a[key]) - k8sStyledAgeToSeconds(b[key])
 }
+
+const highlightText = (highlighted) => `<mark>${highlighted}</mark>`
```
Comment (shivani170) [inline src/components/common/helpers/Helpers.tsx:1211]: Move this const inside `highlightSearchedText`

File: src/components/login/Login.tsx [hunk 3]
```diff
@@ -32,16 +35,16 @@ export default class Login extends Component<LoginProps, LoginFormState> {
 
     componentDidMount() {
         let queryString = new URLSearchParams(this.props.location.search)
```
Comment (milindk8) [inline src/components/login/Login.tsx:37]: make it const

**Review by milindk8 — APPROVED** [review]

**Review by shivani170 — APPROVED** [review]

File: src/components/ResourceBrowser/ResourceList/ResourceList.tsx [hunk 4]
```diff
@@ -416,7 +416,8 @@ export default function ResourceList() {
             if (retainSearched) {
                 handleFilterChanges(searchText, result, true)
             } else {
-                setFilteredResourceList(result.data)
+                const sorted_data = result.data.sort((a, b) => a["name"].localeCompare(b["name"]))
```
Comment (vivek-devtron) [inline src/components/ResourceBrowser/ResourceList/ResourceList.tsx:419]: variable name should be in camelCase

Comment (vivek-devtron) [reply src/components/ResourceBrowser/ResourceList/ResourceList.tsx:419]: Have you verified this for events? Because events don't have name column

Comment (vivek-devtron) [reply src/components/ResourceBrowser/ResourceList/ResourceList.tsx:419]: Is sorting not required in case of retained search?

Comment (dhananjay0601) [reply src/components/ResourceBrowser/ResourceList/ResourceList.tsx]: I have removed the sorting of the result.data there is no requirement of sorting the data.

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**